### PR TITLE
Use parseSimpleArray

### PR DESCRIPTION
**Pull request by 654wak654**
**When merged this pull request will:**
- Use the new [parseSimpleArray](https://community.bistudio.com/wiki/parseSimpleArray) command instead of `call compile` where possible.

### CODE REVIEW

**Comment by jonpas** [comment]
Requires 1.68.

**Comment by 654wak654** [comment]
On second thought, `parseSimpleArray` probably can't parse those addActions. Downloading RC branch.

File: addons/markers/XEH_preInit.sqf [hunk 1]
```diff
@@ -39,7 +39,7 @@ if (isNil QGVAR(MarkerColorsCache)) then {
 
             {
                 if !( _x isEqualType 0) then {
-                    _rgba set [_forEachIndex, call compile _x];
+                    _rgba set [_forEachIndex, parseSimpleArray _x];
```
Comment (PabstMirror) [inline addons/markers/XEH_preInit.sqf]: throws script error, `_x` is not an array
might be able to use on _rgba, unless mixed mode arrays are possible?
e.g. `color[] = {1,0,0,"(profilenamespace getvariable ['Map_Unknown_A',0.8])"};`

Comment (PabstMirror) [inline addons/markers/XEH_preInit.sqf]: oh yeah, `parseSimpleArray` won't be able to handle expressions like `profileNamespace getVar`

**Comment by PabstMirror** [comment]
This command is made for reading machine made strings.
It chokes pretty hard on stuff a human might write

`parseSimpleArray "[- 1.0,2.,3 ,4]"` = [0,nil,nil,4]
`call compile "[- 1.0,2.,3 ,4]"` = [-1,2,3,4]

`parseSimpleArray "[""ace_zeus"", ""ace_attach""]"` = ["ace_zeus",nil]
`call compile "[""ace_zeus"", ""ace_attach""]"` = ["ace_zeus","ace_attach"]

**Comment by commy2** [comment]
You have to sanitize white space first at which point it's probably slower than `call compile`.
Benefit is not being able to cheat. Negative is that it can't handle expressions like `profileNamespace getVariable`.

**Pull request closed without merging**